java-frontend/src/test/files/se/ConditionAlwaysTrueOrFalseCheck.java

```diff
@@ -1716,3 +1779,81 @@ Long woo(boolean b) {
     return myLong;
   }
 }
+class KeyboardFocusManager {
+
+  static void processCurrentLightweightRequests() {
+    KeyboardFocusManager manager = getCurrentKeyboardFocusManager();
+    LinkedList localLightweightRequests = null;
+
+    Component globalFocusOwner = manager.getGlobalFocusOwner();
+    if ((globalFocusOwner != null) &&
+      (globalFocusOwner.appContext != AppContext.getAppContext())) {
+      return;
+    }
+
+    synchronized (heavyweightRequests) {
+      if (currentLightweightRequests != null) {
+        clearingCurrentLightweightRequests = true;
+        disableRestoreFocus = true;
+        localLightweightRequests = currentLightweightRequests;
+        allowSyncFocusRequests = (localLightweightRequests.size() < 2);
+        currentLightweightRequests = null;
+      } else {
+        return;
+      }
+    }
+
+    Throwable caughtEx = null;
+    try {
+      if (localLightweightRequests != null) { // Noncompliant {{Change this condition so that it does not always evaluate to "true"}} this big method requires more than 10000 steps
```

Comment:
It should be possible to considerably reduce this method in order to only contain a concise reproducer.